x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -157,18 +142,17 @@ public JwtRealm(final RealmConfig realmConfig, final SSLService sslService, fina
     }
 
     // must call parseAlgsAndJwksHmac() before parseAlgsAndJwksPkc()
-    private Tuple<List<JWK>, List<String>> parseAlgsAndJwksHmac() {
+    private JwtRealm.JwksAlgs parseJwksAlgsHmac() {
         final SecureString hmacJwkSetContents = super.config.getSetting(JwtRealmSettings.HMAC_JWKSET);
         final SecureString hmacKeyContents = super.config.getSetting(JwtRealmSettings.HMAC_KEY);
+        // HMAC Key vs HMAC JWKSet settings are mutually exclusive
         if (Strings.hasText(hmacJwkSetContents) && Strings.hasText(hmacKeyContents)) {
             throw new SettingsException("HMAC JWKSet and HMAC Key settings are not allowed at the same time.");
```

Comment:
nit: point to the actual setting key, instead of a description of it , in the error message
```suggestion
            throw new SettingsException(RealmSettings.getFullSettingKey(super.config, JwtRealmSettings.HMAC_JWKSET) + " and " + RealmSettings.getFullSettingKey(super.config, JwtRealmSettings.HMAC_KEY)  + "are not allowed at the same time.");
```